### ISSUE TITLE
Default datadir for Linux

### ISSUE THREAD

**Issue by mookie-** [elastic/puppet-elasticsearch#675]
## Feature Description

Why do you use `/usr/share/elasticsearch/data` as default datadir for Linux? I think this is the wrong directory. It should be `/var/lib/elasticsearch/data`.

From: http://www.tldp.org/LDP/Linux-Filesystem-Hierarchy/html/usr.html

```
/usr/share

    This directory contains 'shareable', architecture-independent files (docs, icons, fonts etc). Note, however, that '/usr/share' is generally not intended to be shared by different operating systems or by different releases of the same operating system. Any program or package which contains or requires data that doesn't need to be modified should store that data in '/usr/share' (or '/usr/local/share', if installed locally). It is recommended that a subdirectory be used in /usr/share for this purpose."
```

Also https://www.elastic.co/guide/en/elasticsearch/reference/current/setup-dir-layout.html:

```
Type: data  

Description: The location of the data files of each index / shard allocated on the node.

Location Debian/Ubuntu: /var/lib/elasticsearch/data

Location RHEL/CentOS: /var/lib/elasticsearch
```

**Comment by tylerjl** [comment]
I tend to agree here; keeping data in `/var/lib` is the de facto location that most people would expect to find it.

I tried to look into why this was changed in the module in commit 72c4c76 but didn't find anything obvious. If @electrical is listening in, maybe he can offer some insight into why the defaults are set to `/usr/share` rather than `/var/lib`? There's probably a good reason here that we aren't seeing. 

Regardless, if we do change this it would **have** to be a backwards-incompatible change (i.e. in a major point release, or 1.x as of the time of this writing) since this would change the data directory for existing instances entirely and would be a fairly big change.

Although `/var/lib` is probably the most correct answer, I do want to field input here from other users as well, since the paramount concern is usability and it would definitely be a bad move to shift this out from underneath everyone if the vast majority of people want to keep it as-is.

Thoughts? 💡
